src/providers/mod.rs

```diff
@@ -74,3 +74,50 @@ fn rand_name_sep(prefix: &str, sep: impl Into<Sep>) -> String {
 pub mod aws;
 pub mod azure;
 pub mod baremetal;
+
+fn setup_machine(
```

Comment:
I think it's fine to re-do the parse of the addr for baremetal given how much code it allows us to reuse. Either way is okay though.